cmd/bosun/sched/check.go

```diff
@@ -218,26 +221,33 @@ func (s *Schedule) CheckUnknown() {
 func (s *Schedule) CheckAlert(T miniprofiler.Timer, r *RunHistory, a *conf.Alert) {
 	log.Printf("check alert %v start", a.Name)
 	start := time.Now()
-	var warns expr.AlertKeys
-	crits, err := s.CheckExpr(T, r, a, a.Crit, StCritical, nil)
-	if err == nil {
-		warns, _ = s.CheckExpr(T, r, a, a.Warn, StWarning, crits)
+	var warns, crits expr.AlertKeys
+	ignore := expr.AlertKeys{}
+	d, err := s.ExecuteExpr(T, r, a, a.Depends)
+	if err != nil {
+		collect.Add("check.errs", opentsdb.TagSet{"metric": a.Name}, 1)
+		log.Println(err)
+	} else {
+		for _, dep := range d.Results {
+			ak := expr.NewAlertKey(a.Name, dep.Group)
+			ignore = append(ignore, ak)
```

Comment:
Ok, I figured some stuff out. We need to leave the ignore stuff alone and make a new data structure for this because we need to tell ignore and depends apart, since depends have to mark the alertkey as Unevaluated, but ignore doesn't do that.